File: eBibliotekaServer/LibraryModule/Models/MembershipOffer.cs
using eBibliotekaServer.MembershipModule.Models;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eBibliotekaServer.LibraryModule.Models
{
    public class MembershipOffer
    {
        [Key]
        public int ID { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int AddedMonths { get; set; }
        public int NoOfBooks { get; set; }
        public bool Active { get; set; }
        public int LibraryID { get; set; }
        [ForeignKey(nameof(LibraryID))]
        public Library Library { get; set; }
    }
}

File: eBibliotekaClient/src/app/data/interfaces/MembershipOffer.ts
export class MembershipOffer {
  id: number;
  title: string;
  description: string;
  price: number;
  addedMonths: number;
  noOfBooks: number;
  active: boolean;

  /**
   *
   */
  constructor() {
    this.id = -1;
    this.title = 'Ucitavanje...';
    this.description = 'Ucitavanje...';
    this.price = -1;
    this.addedMonths = -1;
    this.noOfBooks = -1;
    this.active = false;
  }
}

export class MembershipOfferCreateVM {
  title: string;
  description: string;
  price: number;
  addedMonths: number;
  noOfBooks: number;

  /**
   *
   */
  constructor() {
    this.title = '';
    this.description = '';
    this.price = 1;
    this.addedMonths = 1;
    this.noOfBooks = 1;
  }
}

export class MembershipOfferUpdateVM {
  title: string;
  description: string;
  price: number;
  addedMonths: number;
  noOfBooks: number;
  active: boolean;

  /**
   *
   */
  constructor() {
    this.title = 'Ucitavanje...';
    this.description = 'Ucitavanje...';
    this.price = -1;
    this.addedMonths = -1;
    this.noOfBooks = -1;
    this.active = false;
  }
}
